klog.go

```diff
@@ -1265,6 +1276,46 @@ func Infof(format string, args ...interface{}) {
 	logging.printf(infoLog, logging.logr, format, args...)
 }
 
+// InfoS structured logs to the INFO log.
+// The msg argument used to add constant description to the log line.
+// The key/value pairs would be join by "=" ; a newline is always appended.
+//
+// Basic examples:
+// >> klog.InfoS("Pod status updated", "pod", "kubedns", "status", "ready")
+// output:
+// >> I1025 00:15:15.525108       1 controller_utils.go:116] "Pod status updated" pod="kubedns" status="ready"
+func InfoS(msg string, keysAndValues ...interface{}) {
+	b := &bytes.Buffer{}
+	infos(b, logging.logr, msg, keysAndValues)
+
+	logging.println(infoLog, logging.logr, b)
+
+}
+
+func infos(b *bytes.Buffer, logr logr.InfoLogger, msg string, keysAndValues []interface{}) {
+	if logr != nil {
```

Comment:
I don't think this will work as before. logging.println will still be called.

Reply:
Ops, done